### PR TITLE
Cylinder absorp corr

### PR DESCRIPTION
**Pull request by jokasimr**
Fixes https://github.com/scipp/essdiffraction/issues/22

### CODE REVIEW

**Comment by jokasimr** [comment]
@jl-wynen this is ready for review. It is a lot of additions so I imagine it might be easier to go over parts of it together. In that case just tell me and we will set a time

File: docs/user-guide/absorption-correction.ipynb [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "\n",
+    "from scippneutron.absorption import compute_transmission_map\n",
+    "from scippneutron.absorption.cylinder import Cylinder\n",
+    "\n",
+    "\n",
+    "class DummyMaterial:\n",
+    "    'This is probably not a good physical model, but it illustrates the interface'\n",
+    "    def attenuation_coefficient(wavelength):\n",
```
Comment (jl-wynen) [inline docs/user-guide/absorption-correction.ipynb:63]: Change to `def attenuation_coefficient(self, wavelength)` to match the interface of `Material`.

File: src/scippneutron/absorption/cylinder.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+class Cylinder(SampleShape):
+    symmetry_line: sc.Variable
+    center_of_base: sc.Variable
+    radius: sc.Variable
+    height: sc.Variable
+
+    def beam_intersection(
+        self, start_point: sc.Variable, direction: sc.Variable
+    ) -> sc.Variable:
+        'Length of intersection between beam and cylinder'
```
Comment (jl-wynen) [inline src/scippneutron/absorption/cylinder.py:23]:
```suggestion
"""Length of intersection between beam and cylinder"""
```
I'm surprised this isn't caught by ruff.

File: src/scippneutron/absorption/quadratures.py [hunk 3]
```diff
@@ -0,0 +1,996 @@
+disk12 = {
```
Comment (jl-wynen) [inline src/scippneutron/absorption/quadratures.py:1]: Please document what each dict means and how the values were computed.

File: src/scippneutron/absorption/cylinder.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+            quad = _cylinder_quadrature_from_product(
+                quadratures.disk12,
+                dict(x=x, weights=w),  # noqa: C408
+            )
+        elif (
+            kind == 'mc'
+            or isinstance(kind, tuple)
+            and len(kind) > 0
+            and kind[0] == 'mc'
+        ):
```
Comment (jl-wynen) [inline src/scippneutron/absorption/cylinder.py:78]: Try using pattern matching to simply for condition. I don't think you can match both patterns in a single branch. But you could use something like
```python
def foo(x):
    match x:
        case 'low':
            print('low precision')
        case 'mc': # default value for k
            foo(('mc', 5000))
        case ('mc', k):
            print('MC with ', k)
        case _:
            raise ValueError("Bad x")
```

File: docs/user-guide/absorption-correction.ipynb [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    "\n",
+    "def show_correction_map(da):\n",
+    "    'Plotting utility'\n",
+    "    return (\n",
+    "        da['wavelength', 0]['r', 0].plot() /\n",
+    "        da['wavelength', 0]['r', 0]['theta', da.sizes['theta']//2].plot() /\n",
+    "        da['wavelength', 0]['r', 0]['theta', da.sizes['theta']//2]['phi', da.sizes['phi']//4 - da.sizes['phi']//6:da.sizes['phi']//4 + da.sizes['phi']//6].plot() /\n",
+    "        da['wavelength', 0]['r', 0]['theta', da.sizes['theta']//2]['phi', da.sizes['phi']//2 - da.sizes['phi']//6:da.sizes['phi']//2 + da.sizes['phi']//6].plot() /\n",
+    "        da['wavelength', 0]['r', 0]['phi', da.sizes['phi']//2].plot()\n",
+    "    )"
```
Comment (jl-wynen) [inline docs/user-guide/absorption-correction.ipynb:126]: Please add y-axis labels.

File: src/scippneutron/absorption/material.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import scipp as sc
+
+
+@dataclass
+class Material:
+    # attenuation cross section at :math:`\lambda = 1.7982 \mathrm{angstrom}`.
+    attenuation_cross_section: sc.Variable
+    scattering_cross_section: sc.Variable
+    effective_sample_number_density: sc.Variable
```
Comment (jl-wynen) [inline src/scippneutron/absorption/material.py:11]: Use `scn.atoms`. Maybe store an `scn.atoms.ScatteringParams` instance as an attribute of `Material`.

**Review by jl-wynen — COMMENTED** [review]
First high-level review. I will dive into more implementation details later.

File: src/scippneutron/absorption/cylinder.py (hunk 4 above)
Comment (jokasimr) [reply src/scippneutron/absorption/cylinder.py:78]: Maybe it's not worth complicating it in that case. It's clear enough as it is right?

**Comment by SimonHeybrock** [comment]
What is the status here? Is this waiting for scientist input?

File: src/scippneutron/absorption/quadratures.py [hunk 7]
```diff
@@ -1,3 +1,17 @@
+"""Quadrature rules for computing integrals on the unit disk
+For each dictionary :pycode:`{x: ..., y: ..., weights: ...}` in this module:
+:math:`\\int_{||(x, y)|| < 1} f(x, y) dxdy \\approx \\sum_i weights[i] f(x[i], y[i])`
```
Comment (jl-wynen) [inline src/scippneutron/absorption/quadratures.py:3]:
```suggestion
"""Quadrature rules for computing integrals on the unit disk.

For each dictionary ``{x: ..., y: ..., weights: ...}`` in this module:
:math:`\\int_{||(x, y)|| < 1} f(x, y) dxdy \\approx \\sum_i weights[i] f(x[i], y[i])`
```

Empty line to separate brief description from remaining body and render the summary properly.

Double backticks for rendering code.

**Review by jl-wynen — COMMENTED** [review]
Minor formatting adjustment. Otherwise, it looks good.

**Comment by jl-wynen** [comment]
> What is the status here? Is this waiting for scientist input?

It slipped through the cracks. Thanks for reminding me!

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**